pkg/operator/operator.go

```diff
@@ -743,3 +776,244 @@ func (o *Operator) Config(ctx context.Context, key string) (*manifests.Config, e
+		return configv1.ConditionTrue, "", "", nil
+	}
+
+	workloadsCorrectlySpread, reason, message, err := o.WorkloadsCorrectlySpread(ctx)
+	if err != nil {
+		return configv1.ConditionUnknown, "", "", err
+	}
+
+	if !workloadsCorrectlySpread {
+		return configv1.ConditionFalse, reason, message, nil
+	}
+
+	return configv1.ConditionTrue, reason, message, nil
+}
+
+func (o *Operator) ensureNodesAreUncordonned(ctx context.Context) error {
+	for nodeName := range o.nodesCordon {
+		node, err := o.client.GetNode(ctx, nodeName)
+		if apierrors.IsNotFound(err) {
+			delete(o.nodesCordon, nodeName)
+			continue
+		}
+		if err != nil {
+			return err
+		}
+
+		err = drain.RunCordonOrUncordon(o.drainer, node, false)
+		if err != nil {
+			return err
+		}
+		delete(o.nodesCordon, node.Name)
+	}
+	return nil
+}
+
+func (o *Operator) spreadWorkloads(ctx context.Context, namespace string, sel map[string]string) (bool, error) {
+	podList, err := o.client.ListPods(ctx, namespace, metav1.ListOptions{LabelSelector: labels.FormatLabels(sel)})
+	if err != nil {
+		return false, err
+	}
+
+	if len(podList.Items) <= 1 {
+		return false, nil
+	}
+
+	node, err := o.client.GetNode(ctx, podList.Items[0].Spec.NodeName)
+	if err != nil {
+		return false, err
+	}
+
+	err = drain.RunCordonOrUncordon(o.drainer, node, true)
+	if err != nil {
+		return false, err
+	}
+	o.nodesCordon[node.Name] = struct{}{}
+
+	defer func() {
+		err = drain.RunCordonOrUncordon(o.drainer, node, false)
+		if err != nil {
+			klog.Errorf("Couldn't uncordon node %s: err %v", node.Name, err)
+		} else {
+			delete(o.nodesCordon, node.Name)
+		}
+	}()
+
+	var workloadSpread int
+	for _, pod := range podList.Items {
+		for _, vol := range pod.Spec.Volumes {
+			if vol.PersistentVolumeClaim != nil {
+				pvc, err := o.client.GetPersistentVolumeClaim(ctx, pod.Namespace, vol.PersistentVolumeClaim.ClaimName)
+				if apierrors.IsNotFound(err) {
+					continue
+				}
+				if err != nil {
+					return false, err
+				}
+
+				dropPVC, ok := pvc.Annotations[dropPVCAnnotation]
+				if !ok {
+					continue
+				}
+				if dropPVC == "yes" {
+					// Guard from deleting all PVCs to prevent complete data loss.
+					if workloadSpread == len(podList.Items)-1 {
+						// Remove the annotation so that the PVC doesn't get deleted in future cycles.
+						delete(pvc.Annotations, dropPVCAnnotation)
+						_, err := o.client.UpdatePersistentVolumeClaim(ctx, pvc)
+						if err != nil {
+							return false, err
+						}
+						continue
+					}
+
+					err := o.client.DeletePersistentVolumeClaim(ctx, pvc)
+					if err != nil {
+						return false, err
+					}
+					err = o.client.DeletePod(ctx, &pod)
+					if err != nil {
+						return false, err
+					}
+
+					workloadSpread++
+				}
+			}
+		}
+	}
+
+	return workloadSpread > 0, nil
+}
+
+// workloadCorrectlySpread returns whether the selected pods are spread across
+// different nodes ensuring proper high-availability.
```

Comment:
```suggestion
// workloadCorrectlySpread returns whether the selected pods are spread across
// different nodes ensuring proper high-availability.
// If the pods don't use persistent storage, it will always return true.
```